src/Components/Users/UsersListTable.js

```diff
@@ -0,0 +1,276 @@
+    page > 0 ? Math.max(0, (1 + page) * rowsPerPage - usersList.length) : 0;
+
+  const deleteUser = (id) => {
+    const newUsersList = usersList.filter((user) => user.id !== id);
+    const total = usersList.length - 1;
+    const pages = total / rowsPerPage;
+
+    setUsersList(newUsersList);
+
+    if (page === pages && page !== 0) {
+      setPage(page - 1);
+    }
+  };
+
+  return (
+    <Container sx={{ my: '1rem' }}>
+      <Box>
+        <Paper>
+          <Toolbar>
+            <Typography
+              component="div"
+              id="tableTitle"
+              sx={{ mr: 'auto' }}
+              variant="h6">
+              Lista de Usuarios
+            </Typography>
+            <Button
+              component={Link}
+              to="/backoffice/users/create"
+              variant="contained">
+              Nuevo Usuario
+            </Button>
+          </Toolbar>
+          <TableContainer>
+            <Table
+              aria-labelledby="tableTitle"
+              size="small"
+              sx={{ minWidth: 600 }}>
+              <TableHead>
+                <TableRow>
+                  <TableCell sortDirection={orderBy === 'name' ? order : false}>
+                    <TableSortLabel
+                      active={orderBy === 'name'}
+                      direction={orderBy === 'name' ? order : 'asc'}
+                      onClick={createSortHandler('name')}>
+                      Nombre
+                      {orderBy === 'name' ? (
+                        <Box component="span" sx={visuallyHidden}>
+                          {order === 'desc'
+                            ? 'ordenado descendente'
+                            : 'ordenado ascendente'}
+                        </Box>
+                      ) : null}
+                    </TableSortLabel>
+                  </TableCell>
+                  <TableCell
+                    align="right"
+                    sortDirection={orderBy === 'email' ? order : false}>
+                    <TableSortLabel
+                      active={orderBy === 'email'}
+                      direction={orderBy === 'email' ? order : 'asc'}
+                      onClick={createSortHandler('email')}>
+                      Email
+                      {orderBy === 'email' ? (
+                        <Box component="span" sx={visuallyHidden}>
+                          {order === 'desc'
+                            ? 'ordenado descendente'
+                            : 'ordenado ascendente'}
+                        </Box>
+                      ) : null}
+                    </TableSortLabel>
+                  </TableCell>
+                  <TableCell align="right">Acciones</TableCell>
+                </TableRow>
+              </TableHead>
+              <TableBody>
+                {usersList
+                  .sort(getComparator(order, orderBy))
+                  .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
+                  .map((row) => {
```

Comment:
Please remove this logic because there should not be code logic like this inside html. Here you should already have the list filtered and sorted, ready yo be used.